src/sentry/search/events/types.py

```diff
@@ -118,5 +118,6 @@ class QueryBuilderConfig:
     # of a top events request
     skip_tag_resolution: bool = False
     on_demand_metrics_enabled: bool = False
+    on_demand_metrics_type: Optional[Any] = (None,)
```

Comment:
If we import here it causes a circular import somewhere, I didn't have time to hunt it down.

Reply:
Just going to leave this since we really need to restart extraction for this case, left a ticket on our backlog to fix it if we don't remove MetricSpecType.